File: samples/Python/pangram-checker.py
import string, sys
if sys.version_info[0] < 3:
    input = raw_input

def ispangram(sentence, alphabet=string.ascii_lowercase):
    alphaset = set(alphabet)
    return alphaset <= set(sentence.lower())

print ( ispangram(input('Sentence: ')) )

File: samples/C++/pangram-checker.cpp
#include <algorithm>
#include <cctype>
#include <string>
using namespace std;

const string alphabet("abcdefghijklmnopqrstuvwxyz"); // sorted, no duplicates

bool is_pangram(string s) {
  // Convert to lower case.
  transform(s.begin(), s.end(), s.begin(), ::tolower);
  // Convert to a sorted sequence of (not necessarily unique) characters.
  sort(s.begin(), s.end());
  // Is the second sequence a subset of the first sequence?
  // Repeated letters in "s" are okay, since it still "includes" the single letter
  return includes(s.begin(), s.end(), alphabet.begin(), alphabet.end());
}
